app/src/main/java/org/efidroid/efidroidmanager/LokiTool.java

```diff
@@ -0,0 +1,73 @@
+package org.efidroid.efidroidmanager;
+
+import android.content.Context;
+import android.os.Build;
+
+import org.apache.commons.io.FileUtils;
+import org.efidroid.efidroidmanager.services.IntentServiceEx;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class LokiTool {
+    public enum PartitionLabel {
+        BOOT("boot"),
+        RECOVERY("recovery");
+
+        private final String mLabel;
+
+        PartitionLabel(String s) {
+            mLabel = s;
+        }
+
+        public String getLabel() {
+            return mLabel;
+        }
+    }
+
+    private LokiTool() {
+    }
+
+    public static void init(Context context) {
+        ArrayList<String> abis = new ArrayList<>();
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            abis.addAll(Arrays.asList(Build.SUPPORTED_ABIS));
+        } else {
+            abis.add(Build.CPU_ABI);
+            abis.add(Build.CPU_ABI2);
+        }
+        // do not initialize on non-arm cpu
+        if (!abis.contains("armeabi-v7a")) return;
+
+        try {
+            InputStream is = null;
+            try {
+                is = context.getAssets().open("armeabi-v7a/loki_tool_static");
```

Comment:
this binary isn't statically linked. Linking it dynamically actually isn't such a bad idea because it makes the file size much smaller. But what about compatibility? we need to support 4.0+.
Also, is it possible to build binaries with gradle+ndk? The only reason I'm using busybox prebuilt is that it doesn't work that well when compiled using the ndk and still needs to be build using autoconf anyway.

Either way, we don't include the link-type in the filename.